### ISSUE TITLE
TOCO is currently not working from command line

### ISSUE THREAD

**Issue by codeteo** [googlecodelabs/tensorflow-for-poets-2#52]
OS Version : Ubuntu 16.04
Tensorflow Version: 1.8.0

When I run the command `toco --help` the output is : 

```
TOCO from pip install is currently not working on command line.
Please use the python TOCO API or use
bazel run tensorflow/contrib/lite:toco -- <args> from a TensorFlow source dir.
```

Is this because I'm using Tensorflow 1.8 or something else is wrong and what should I do about it ?

**Comment by thanhluan001** [comment]
Yeah, it's true that tensorflow 1.8 got the problem with toco. I downgraded to 1.7 and it seems to work. 

`pip install --upgrade  "tensorflow==1.7.*"`

Ref: https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2-tflite/index.html#1

**Comment by raibosome** [comment]
Alternatively, if you want to better manage different versions of TensorFlow in your machine, you can create a new conda environment just for TensorFlow 1.7.

```
conda create --name tensorflowlite python=3.6
conda activate tensorflowlite
pip install tensorflow==1.7.*
```

Then try running `toco` again:
```
toco --help
```

Just remember to activate this environment every time you want to run TF with version 1.7 with `conda activate tensorflowlite`, and `conda deactivate` once you're done.

**Comment by Sagarsharma4244** [comment]
Toco is working in Tensorflow1.9 Just update it

**Comment by AliceDinh** [comment]
@Sagarsharma4244 and others
With TF 1.8.0, I met the error:
`ModuleNotFoundError: No module named 'tensorflow.contrib.lite.toco.python'`
I updated TF to 1.9.0 as your suggestion but the error is:
`ModuleNotFoundError: No module named 'tensorflow.contrib.lite.python.tflite_convert'`
My system: Window 10, TF is installed from pip: pip install tensorflow

**Comment by alensaqe** [comment]
You can upgrade to Tf 1.9+ and use TensorFlow Lite Optimizing Converter command-line as below:

```
tflite_convert \
  --output_file=/tmp/foo.tflite \
  --saved_model_dir=/tmp/saved_model
```

Docs at : [https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/cmdline_examples.md](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/cmdline_examples.md)

**Comment by jason-klein** [comment]
**WORKAROUND**

I encountered the exact same error when running `toco` on macOS, even after setting up Python 3.6 in a conda environment.

```
$ python -V
Python 3.6.6 :: Anaconda, Inc.
```

```
$ toco
TOCO from pip install is currently not working on command line.
Please use the python TOCO API or use
bazel run tensorflow/contrib/lite:toco -- <args> from a TensorFlow source dir.
```

I used `which` on macOS to locate the `toco` executable on macOS. _(Use `whereis` on Linux)_

```
$ which toco
/usr/local/bin/toco
```

I realized this is a Python script with the very first line hard-coded to interpret the script using Python 2.7. , but my conda environment was using Python 3.6. 

```
$ head -n1 /usr/local/bin/toco
#!/usr/local/opt/python@2/bin/python2.7
```

This hard-coded Python version was causing the error, so I forced the script to use Python 3.6 from my environment. Normal output, instead of an error!

```
$ python /usr/local/bin/toco
usage: /usr/local/bin/toco
Flags:
...
```

I was able to successfully convert a TensorFlow model to TensorFlow Lite by substituting the `toco` command with `python /usr/local/bin/toco` followed by all of the normal toco parameters.

Sharing because I would expect others on macOS and Linux to be able to use this same workaround to bypass the "TOCO from pip install is currently not working on command line" error.

**Comment by Satoshinakamura00** [comment]
For those who still think its possible to convert your models to tflite you cant. no matter what model format you used on whatever platform the conversion is deemed impossible because the main reason is that all tensor flow dependencies are flawed and broken, Toco doesnt work so dont use it , trying to use it is a waste of time. 

Also no offence Klein but if your method truly worked, why not display the whole code along with the returning statement in terminal stating the conversion was a success instead of just showing this " flags:.... and nothing else " , im sorry to say this but the fact you show no proof of the conversion being successful, i find it hard to believe that it can be actually be used as a workaround.

For those who want to use Tensorflow in the near future , dont use it at any cost because it'll just burn you to death since its a broken machine leaning platform , the training aspect of the algorithm works but the latter such as converting to tflite and other factors other packages dont work at all so pleas dont waste your time.
